wrappers/services.go

```diff
@@ -288,3 +342,61 @@ WantedBy={{.ServicesTarget}}
 
 	return templateOut.Bytes()
 }
+
+func genSocketFile(appInfo *snap.AppInfo, socketName string) []byte {
+	socketTemplate := `[Unit]
+# Auto-generated, DO NO EDIT
+Description=Socket {{.SocketName}} for snap application {{.App.Snap.Name}}.{{.App.Name}}
+Requires={{.MountUnit}}
+Wants={{.PrerequisiteTarget}}
+After={{.MountUnit}} {{.PrerequisiteTarget}}
+X-Snappy=yes
+
+[Socket]
+Service={{.ServiceFileName}}
+ListenStream={{.SocketInfo.ListenStream}}
+{{if .SocketInfo.SocketMode}}SocketMode={{.SocketInfo.SocketMode | printf "%04o"}}{{end}}
```

Comment:
This also needs to include FileDescriptorName, which is pretty much the only real reason to have a high-level name for those.

Reply:
Done